src/parser.ts

```diff
@@ -256,6 +258,12 @@ export class Parser {
         this.errorHandler.tolerate(this.unexpectedTokenError(token, message));
     }
 
+    tolerateInvalidLoopStatement() {
```

Comment:
Had to create separate function as for statement parsing started to fail due to cyclomatic complexity